index.js

```diff
@@ -253,18 +284,21 @@ Object.defineProperty(File.prototype, 'extname', {
 
 Object.defineProperty(File.prototype, 'path', {
   get: function() {
-    return this.history[this.history.length - 1];
+    var last = this.history[this.history.length - 1];
+    return this.isDirectory() ? endWithSep(last) : last;
```

Comment:
Yeah, makes sense. I placed it into a getter since you can't guarantee that `this.isDirectory()` will be static. But that can be maybe handled by documenting that switch between file and directory should be only made in `stat` option passed to constructor.